pandas/core/groupby/generic.py

```diff
@@ -559,7 +559,6 @@ def _transform_general(
                 result = maybe_downcast_numeric(result, self._selected_obj.dtype)
 
         result.name = self._selected_obj.name
-        result.index = self._selected_obj.index
```

Comment:
We only want to remove this line when going through _set_result_index_ordered above. Instead, move it to the else clause a few lines up.

Reply:
so the else clause is when we're returning an empty object (for a non-empty result we pass through _set_result_index_ordered)

Reply:
Ah, makes sense. Thanks.